### PR TITLE
Test Scp and Ssh

### PR DESCRIPTION
**Pull request by prudmirya**
Test Scp and Ssh

test scp with copy a 1GB file into peer

Signed-off-by: prudhvi <mprudhvi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by prudmirya** [comment]
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/555943/job.txt)
log added

File: io/net/scp_ssh.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    def testscpandssh(self):
+        '''
+        check scp and ssh
+        '''
+        cmd = "ssh %s@%s echo hi" % (self.user, self.peer)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("unable to ssh into peer machine")
+        process.run("dd if=/dev/zero of=/tmp/tempfile bs=1024000000 count=1",
+                    shell=True)
+        time.sleep(15)
```
Comment (abdhaleegit) [inline io/net/scp_ssh.py]: is sleep is necessary here ?

File: io/net/scp_ssh.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        cmd = "timeout 600 scp /tmp/tempfile %s@%s:/tmp" %\
+              (self.user, self.peer)
+        ret = process.system(cmd, shell=True, verbose=True)
+        time.sleep(15)
+        if ret != 0:
+            self.fail("unable to copy into peer machine")
+        cmd = "timeout 600 scp %s@%s:/tmp/tempfile /tmp" %\
+              (self.user, self.peer)
+        ret = process.system(cmd, shell=True, verbose=True)
+        time.sleep(15)
```
Comment (abdhaleegit) [inline io/net/scp_ssh.py]: and here ?

File: io/net/scp_ssh.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        for pkg in ["openssh-clients", "net-tools"]:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+        self.peer = self.params.get("peerip")
+        self.user = self.params.get("user_name")
+
```
Comment (abdhaleegit) [inline io/net/scp_ssh.py]: not mandatory, but looks good in code : '/tmp/tempfile' is used 6 times in code, why not initilazed as self.tmpfile = '/tmp/tempfile'

File: io/net/scp_ssh.py.data/README.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Description:
+------------------------
+This Program to check both ssh and scp.
+-----------------------------
```
Comment (abdhaleegit) [inline io/net/scp_ssh.py.data/README.txt]: Please explain more about what this test do

File: io/net/scp_ssh.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    '''
+    check the ssh into peer
+    check the scp into peer
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        for pkg in ["openssh-clients", "net-tools"]:
```
Comment (abdhaleegit) [inline io/net/scp_ssh.py]: distro's may have different package name is it handled ? ignore this comment if "openssh-clients", "net-tools" package names are common in all distributions ?

File: io/net/scp_ssh.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            self.fail("unable to copy into peer machine")
+        cmd = "timeout 600 scp %s@%s:/tmp/tempfile /tmp" %\
+              (self.user, self.peer)
+        ret = process.system(cmd, shell=True, verbose=True)
+        time.sleep(15)
+        if ret != 0:
+            self.fail("unable to copy from peer machine")
+        md_val2 = hashlib.md5(open('/tmp/tempfile', 'rb').read()).hexdigest()
+        time.sleep(5)
+        if not md_val1 == md_val2:
```
Comment (narasimhan-v) [inline io/net/scp_ssh.py]: it can be `if md_val1 != md_val2:`

File: io/net/scp_ssh.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    def tearDown(self):
+        '''
+        remove data both peer and host machine
+        '''
+        self.log.info('removing data')
+        cmd = "rm /tmp/tempfile"
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("unable to remove data from client")
+        msg = "rm /tmp/tempfile"
+        cmd = "ssh %s \"%s\"" % (self.peer, msg)
```
Comment (narasimhan-v) [inline io/net/scp_ssh.py]: it should be `cmd = "ssh %s@%s \"%s\"" % (self.user, self.peer, msg)`

File: io/net/scp_ssh.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+        self.peer = self.params.get("peerip")
+        self.user = self.params.get("user_name")
+
+    def testscpandssh(self):
+        '''
+        check scp and ssh
+        '''
+        cmd = "ssh %s@%s echo hi" % (self.user, self.peer)
```
Comment (narasimhan-v) [inline io/net/scp_ssh.py]: `cmd = "ssh %s@%s \"echo hi\"" % (self.user, self.peer)`

File: io/net/scp_ssh.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        for pkg in ["openssh-clients", "net-tools"]:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+        self.peer = self.params.get("peerip")
+        self.user = self.params.get("user_name")
```
Comment (PraveenPenguin) [inline io/net/scp_ssh.py]: can you please pass any default for value `peerip` and  `user_name `or check if multiplexer input not provided and test should fail ?

File: io/net/scp_ssh.py (hunk 2 above)
Comment (prudmirya) [inline io/net/scp_ssh.py]: yes sleep is need, otherwise terminated by sigterm event,

File: io/net/scp_ssh.py.data/scp_ssh.yaml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+Parameters:
+    peerip: "9.40.195.85"
```
Comment (prudmirya) [inline io/net/scp_ssh.py.data/scp_ssh.yaml]: u mean peer interface is need to pass?

File: io/net/scp_ssh.py (hunk 6 above)
Comment (prudmirya) [inline io/net/scp_ssh.py]: ok sure

File: io/net/scp_ssh.py (hunk 7 above)
Comment (prudmirya) [inline io/net/scp_ssh.py]: ok, sure

File: io/net/scp_ssh.py (hunk 8 above)
Comment (prudmirya) [inline io/net/scp_ssh.py]: ok sure

File: io/net/scp_ssh.py.data/scp_ssh.yaml (hunk 10 above)
Comment (narasimhan-v) [inline io/net/scp_ssh.py.data/scp_ssh.yaml]: @prudmirya @abdhaleegit means to say remove the IP from yaml file when upstreaming the code

**Comment by prudmirya** [comment]
closed the current PR and opened new PR #137

**Pull request closed without merging**